src/Citeproc/Types.hs

```diff
@@ -1454,28 +1466,42 @@ instance Uniplate (Output a) where
 instance Biplate (Output a) (Output a) where
   biplate = plateSelf
 
+-- | Describes how a hyperlink relates to a bibitem.
+data LinkType = 
+      LinkNormal  -- ^ an unspecified hyperlink
+    | LinkURL     -- ^ a raw URL with no label
+    | LinkTitle   -- ^ hyperlinked title
+    | LinkBibItem -- ^ hyperlinked bib item
+  deriving (Show, Eq)
+
 data Identifier =
       IdentDOI Text
     | IdentPMCID Text
     | IdentPMID Text
     | IdentURL Text
   deriving (Show, Eq)
 
-identifierToUrl :: Identifier -> Text
-identifierToUrl ident =
+identifierToURL :: Identifier -> Text
```

Comment:
identifierToURL is adapted from pandoc's linkify function, and I tried to preserve the same behavior